src/fauxtoshop.cpp

```diff
@@ -371,23 +336,25 @@ void applyScatterFilter(Grid<int> &original){
  * Returns: nothing. Void function
 */
 
-void applyEdgeDetectionFilter(const Grid<int> &original, Grid<int> &blackWhiteGrid){
+void applyEdgeDetectionFilter(GBufferedImage &img){
     //Prompt user for scatter radius input as a positive integer
-    int edgeThreshold = validateUserInput("Enter the edge threshold as a positive number: ", 0, INT_MAX);
+    int edgeThreshold = getValidInteger("Enter the edge threshold as a positive number: ", 0, INT_MAX);
+    //Set up a new image grid for the black and white output for edge detection
+    Grid<int> original = img.toGrid();
+    Grid<int> blackWhiteGrid(original.numRows(), original.numCols());
     //Iterate through each pixel
     for (int i = 0; i < original.numRows(); i++){
         for (int j = 0; j < original.numCols(); j++) {
             //Define boolean to set whether each pixel is an edge or not
             bool isEdge = false;
             //Check each pixel bordering the pixel at relative location (0,0) to detect the edge
-            for(int relative_i = -1; relative_i < 2; relative_i++){
-                for(int relative_j = -1; relative_j <2; relative_j++){
-                    if (!isEdge && original.inBounds(i + relative_i, j + relative_j)){
+            for(int rowOffset = -1; rowOffset < 2; rowOffset++){

```

Comment:
https://web.stanford.edu/class/archive/cs/cs106b/cs106b.1182/handouts/styleguide.html#naming
> **Capitalization:** Name variables and functions with camel-casing **likeThis**, name classes with Pascal casing **LikeThis**, and name constants in uppercase **LIKE_THIS**.

The names `relataveI` and `relataveJ` look a bit strange, so I suggested the names `rowOffset` and `colOffset` instead